File: CCp/C++/prime.cpp

#include <iostream>

bool isPrime( int number ) {
   
    if( number <= 1 ) 
      return false;

    if( number <= 3 ) 
      return true;  
    
    if( number % 2 == 0 || number % 3 == 0 )
       return false;

     for( int i = 5; i * i <= number; i += 6 ) {
       
        if( number % i == 0 ) 
            return false;
     }  

     return true;
}

int main () {
   
    int number;
    std::cout << "Enter the number" << std::endl;
    std::cin >> number;

    std::cout << "Prime numbers upto " << number << " are : ";

    for ( int i = 2; i <= number; i++ ) {
         
         if( isPrime( i ))
            std::cout << i << " ";
    }
    std::cout << std::endl;
}

File: python/variables/prime.py
# print prime numbers upto given input

num = int( input("enter a number : "))

if num < 2:
  print("no prime numbers found")

for n in range(2, num + 1):
  for x in range(2, n):
    if n % x == 0:
      print(n, " equals ", x, "*", n // x)
      break
  else:  
    print(n, " is a prime number")

